### PR TITLE
implement onOpen an onClose in delegate

### PR DESCRIPTION
**Pull request by waelba**

### CODE REVIEW

**Comment by jarthod** [comment]
Do we need to add this even if we're not using them? will out clients need to do this? is it the same for `dimeloChatMessageSendFail`?

**Comment by waelba** [comment]
no `dimeloChatMessageSendFail` will be called every time when message can't be sent (internet down ...) but it's deprecated because it is used before doing the offline mode so we can remove it from demo

**Comment by jarthod** [comment]
What I mean is: does it work if we don't define these methods?

**Comment by waelba** [comment]
yes it works

**Review by jarthod — APPROVED** [review]
Ok cool

File: app/src/main/java/com/dimelo/sampleapp/MainActivity.java [hunk 1]
```diff
@@ -48,17 +50,15 @@ public void onCreate(Bundle savedInstanceState) {
...
-
-            String message = "An error occurred";
-            if (error.statusCode == DimeloConnection.DimeloError.NO_CONNECTION_ERROR) {
-                message = "Please check your Internet connection and try again later.";
-            } else if (error.statusCode == DimeloConnection.DimeloError.TIMEOUT_ERROR) {
-                message = "The server is not responding, please try again later";
-            }
-            Toast.makeText(MainActivity.this, message, Toast.LENGTH_SHORT).show();
+        public void onOpen(Dimelo dimelo) {
+            super.onOpen(dimelo);
```
Comment (jarthod) [inline app/src/main/java/com/dimelo/sampleapp/MainActivity.java]: why do you need to call super here?

**Comment by DrissTM** [comment]
Is this PR still relevant or can we close it ?

**Comment by waelba** [comment]
I will squash it

**Pull request merged**